content/dev/blog/tidyeval.Rmd

```diff
@@ -0,0 +1,232 @@
+title = "Making Graphs Look Easy with ggplot2 and Tidy Evaluation"
+weight = 20
+images = ["https://crunch.io/img/logo-1200x630.png"]
++++
+
+
+```{r code to generate lead plots, echo = FALSE}
+autoplot <- function(df) {
+    # Add grouping variable which was stripped by summarize
+    df <- df %>%
+        group_by(!!!groups(df), !!sym(names(df)[length(groups(df)) + 1]))
+
+    groups <- groups(df)
+    if (length(groups) == 1) {
+        plot_fun <- plot_1d
+    } else {
+        plot_fun <- plot_2d
+    }
+    vars <- syms(names(df))
+    out <- plot_fun(df, vars)
+
+    if (length(groups) > 2) {
+        groups <- syms(groups)
+        out <- out +
+            facet_wrap(vars(!!!groups[3:length(groups)]))
+    }
+    return(out)
+}
+
+plot_1d <- function(df, vars){
+    groups <- vars[vars %in% groups(df)]
+    measure <- vars[length(vars)][[1]]
+    df %>%
+        select(!!groups[[1]], !!measure) %>%
+        arrange(desc(!!measure)) %>%
+        ggplot(aes(x = !!measure, y = !!groups[[1]])) +
+        geom_point() +
+        theme_minimal()
+}
+
+plot_2d <- function(df, vars){
+    groups <- vars[vars %in% groups(df)]
+    measure <- vars[length(vars)][[1]]
+    df %>%
+        select(!!!groups, !!measure) %>%
+        arrange(desc(!!measure)) %>%
+        ggplot(aes(x = !!measure, y = !!groups[[1]], , color = !!groups[[2]])) +
+        geom_point() +
+        theme_minimal()
+}
+```
+
+At Crunch, one of the ways we try to make data exploration simple is by providing sensible default views that take into account the properties of your data and metadata. We've just released some plotting methods in our `crplyr` R package that define methods for `ggplot2`'s [`autoplot()`](https://ggplot2.tidyverse.org/reference/autoplot.html) function.
+`autoplot()` was the ideal approach for us to encapsulate the logic of how to "just do the right thing." You can do the analysis you want, and it will make smart choices about how to display it with no additional input--all of which you can control or override with additional `ggplot2` layers, if you want.
+
+This lets us do [insert examples here, with pictures]
+
+Making it look easy, though, can be hard work. Our `autoplot` methods inspect the input objects to understand their dimensionality and data types and choose an appropriate visualization. Figuring out and passing the right arguments to the right places can be messy, so make these functions work, we took advantage of "[tidyeval](https://dplyr.tidyverse.org/articles/programming.html)", a new framework that systematizes non-standard evaluation in R.
+
+To illustrate this pattern, this blog post goes through a simplified example using the "diamonds" example dataset and pure `dplyr` methods. We'll create a plotting function that adjusts to the number of grouping variables in a tibble. This lets you pipe data from a dplyr pipeline into a single function and get a meaningful, appropriate plot, like this:
+
+```{r}
```

Comment:
If we include real Crunch examples, I think we'll want to skip these examples here. We'll show them in the text below.